charmhelpers/contrib/openstack/files/check_openvswitch.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# -*- coding: us-ascii -*-
+"""Check for OVS interface errors."""
+
+import argparse
+import json
+import subprocess
+import sys
+
+# dependence on nagios_plugin3 being installed by charm-nrpe
+from nagios_plugin3 import CriticalError, UnknownError, try_check
+
+
+def parse_ovs_status():
+    """Check for errors in 'ovs-vsctl show' output."""
+    try:
+        cmd = ["/usr/bin/sudo", "/usr/bin/ovs-vsctl", "--format=json", "list", "Interface"]
+        ovs_output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        raise UnknownError(
+            "UNKNOWN: Failed to query ovs: {}".format(
+                e.output.decode(errors="ignore").rstrip()
+            )
+        )
+    ovs_interfaces = json.loads(ovs_output.decode(errors="ignore"))
+    ovs_interface_errors = []
+    for i in ovs_interfaces["data"]:
+        # OVSDB internal data is formatted per RFC 7047 5.1
+        iface = dict(zip(ovs_interfaces["headings"], i))
+        error = iface["error"]
+        if isinstance(error, list) and len(error) == 2 and error[0] == "set":
+            # deserialize the set data into csv string elements
+            error = ",".join(error[1])
+        if error:
+            ovs_interface_errors.append("Error on iface {}: {}".format(iface["name"], error))
+
+    if ovs_interface_errors:
+        numerrs = len(ovs_interface_errors)
+        raise CriticalError(
+            "CRITICAL: Found {} error(s) in OVSDB: "
+            "{}".format(numerrs, ", ".join(ovs_interface_errors))
+        )
+
+    print("OK: no errors found in openvswitch")
```

Comment:
Perhaps 'no interface errors'?

Reply:
As a followup thought; I always love nagios checks which have a side effect of giving you other information.  For example wouldn't it be cool it said:

OK: no errors found in NN interfaces in Open vSwitch

And the cost would be a simple counter?